protocol/storage/client.go

```diff
@@ -122,7 +119,7 @@ func (smc *Client) ProposeDeal(ctx context.Context, miner address.Address, data
 		return nil, fmt.Errorf("piece is %d bytes but sector size is %d bytes", pieceSize, maxUserBytes)
 	}
 
-	pieceReader, err := smc.api.DAGCat(ctxSetup, data)
+	pieceReader, err := smc.api.DAGCat(ctx, data)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to make piece reader")
 	}
```

Comment:
Could you drop a comment below that `GeneratePieceCommitment` is computationally expensive and could take minutes for large pieces?